plugins/reference-assets/components/src/index.tsx

```diff
@@ -39,6 +52,17 @@ export type DSLSchema = PlayerDSLSchema<
   DataTypeReference<typeof dataTypes, ValidatorFunctionRefs<typeof validators>>
 >;
 
+/** An object wrapper that tracks it's index from a parent */
+const ObjectWithIndexTracking = (props: WithChildren) => {
```

Comment:
Same comment as `WithApplicability` this should also come from the DSL package since we provide the already `OptionalIDSuffixProvider`